modules/osdk-control-compat.adoc

```diff
@@ -0,0 +1,87 @@
+= Controlling Operator compatibility with {product-title} versions
+
+[IMPORTANT]
+====
+Kubernetes periodically deprecates certain APIs that are later removed in subsequent releases. As a result, your Operator project will be unable to use any removed APIs starting with the version of {product-title} that uses whichever Kubernetes version removed the API.
+
+As an Operator author, it is strongly recommended that you review the link:https://kubernetes.io/docs/reference/using-api/deprecation-guide/[Deprecated API Migration Guide] in Kubernetes documentation and ensure that your projects stay updated to avoid using deprecated and removed APIs, ideally before the release of a future version of {product-title} that would make the Operator incompatible.
+====
+
+When an API is removed from an {product-title} version, Operators running on that cluster version that are still using any removed APIs will no longer work properly. As an Operator author, you should plan to update your Operators to accommodate API deprecation and removal to avoid interruptions for users of your Operator.
+
+[TIP]
+====
+You can check the event alerts of your Operators running on {product-title} 4.8 and later to find whether there are any warnings about APIs currently in use. The following alerts fire when they detect an API in use that will be removed in the next release:
+
+`APIRemovedInNextReleaseInUse`::
+APIs that will be removed in the next OpenShift Container Platform release.
+
+`APIRemovedInNextEUSReleaseInUse`::
+APIs that will be removed in the next OpenShift Container Platform link:https://access.redhat.com/support/policy/updates/openshift#ocp4_phases[Extended Update Support (EUS)] release.
+====
+
+To prevent workflow issues, cluster administrators must install a version of your Operator in their cluster that is compatible with the next version of {product-title} before they upgrade their cluster from any previous version. While it is recommended that you update your Operators to no longer use deprecated or removed APIs, if you still need to publish your Operator bundles with removed APIs for continued use on earlier versions of {product-title}, ensure that the bundle is configured accordingly.
+
+The following procedure helps prevent administrators from installing versions of your Operator on an incompatible version of {product-title}.  These steps also prevent administrators from upgrading to a newer version of {product-title} that would be incompatible with the version of your Operator that is currently installed on their cluster.
+
+[IMPORTANT]
+====
+You should configure the following settings in your Operator project as soon as possible. Users running workloads with deprecated APIs that are planning to eventually upgrade to a future version of {product-title}, which might remove the deprecated API, must be running an Operator that has the proper configuration to prevent the cluster upgrade and potentially adversely impact their critical workloads.
+====
```

Comment:
Does it lose much if you remove ", which might remove the deprecated API,"?  

It was jarring (to this newb) to read "...configuration to prevent the cluster upgrade..."  because that never sounds good.  Maybe "...configuration to prevent a cluster upgrade to an incompatible version and..."  Or maybe s/prevent/avoid/?

Reply:
✅
✅